# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kterm/kterm-3dsc-linux.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -uNr KTerm.ad KTerm.ad
--- ./KTerm.ad	Sun Jun 23 10:00:22 1996
+++ ./KTerm.ad	Sat May  4 10:06:44 2002
@@ -123,3 +123,6 @@
 *tek4014*font3: 6x13
 *tek4014*fontSmall: 6x10
 
+*scrollBar:			true
+*Scrollbar.background:		gray
+
diff -uNr kterm.h kterm.h
--- ./kterm.h	Fri Jul 12 07:01:37 1996
+++ ./kterm.h	Sat May  4 10:06:44 2002
@@ -36,6 +36,6 @@
 #define KTERM_KINPUT2	/* Kinput2 protocol */
 #define KTERM_COLOR	/* color sequence */
 #define KTERM_NOTEK	/* disables Tektronix emulation */
-#undef  KTERM_XAW3D	/* Xaw3d -DARROW_SCROLLBAR support */
+#define KTERM_XAW3D	/* Xaw3d -DARROW_SCROLLBAR support */
 
 #endif /* !_KTERM_H_ */
diff -uNr main.c main.c
--- ./main.c	Fri Jul 12 07:01:38 1996
+++ ./main.c	Sat May  4 10:06:44 2002
@@ -140,6 +140,7 @@
 #define HAS_UTMP_UT_HOST
 #define LASTLOG
 #define WTMP
+#include <termios.h>
 #endif
 
 #include <sys/ioctl.h>
diff -uNr screen.c screen.c
--- ./screen.c	Sun Jun 23 10:00:26 1996
+++ ./screen.c	Sat May  4 10:06:44 2002
@@ -46,8 +46,9 @@
 #include <sys/ptem.h>
 #endif
 
-extern Char *calloc(), *malloc(), *realloc();
-extern void free();
+#include <stdlib.h>
+/* extern Char *calloc(), *malloc(), *realloc();
+   extern void free(); */
 
 ScrnBuf Allocate (nrow, ncol, addr)
 /*
diff -uNr scrollbar.c scrollbar.c
--- ./scrollbar.c	Fri Jul 12 07:01:39 1996
+++ ./scrollbar.c	Sat May  4 10:06:44 2002
@@ -29,6 +29,7 @@
 #include "ptyx.h"		/* gets Xt headers, too */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <ctype.h>
 #include <X11/Xatom.h>
 
@@ -324,7 +325,7 @@
 	register TScreen *screen = &xw->screen;
 	register int border = 2 * screen->border;
 	register int i;
-	Char *realloc(), *calloc();
+	/* Char *realloc(), *calloc(); */
 
 	if(screen->scrollbar)
 		return;
